cpp/arcticdb/entity/native_tensor.hpp

```diff
@@ -107,10 +107,11 @@ struct NativeTensor {
         util::warn(strides_condition,
                    "Cannot safely ptr_cast to type of size {} when strides ({}) is not a multiple of elsize ({}) in NativeTensor with dtype {}",
                    sizeof(T), strides_[0], elsize_, data_type());
+        ssize_t signed_pos = pos;
```

Comment:
`ssize_t` is not just a signed version of `size_t`:
https://stackoverflow.com/questions/15739490/should-i-use-size-t-or-ssize-t
You probably want `int64_t` in this case

Reply:
Nice catch! Changed to `int64_t`.